src/element/textElement.ts

```diff
@@ -240,15 +240,13 @@ export const handleBindTextResize = (
       baseline: nextBaseLine,
     });
 
-    if (!isArrowElement(container)) {
-      mutateElement(
-        textElement,
-        computeBoundTextPosition(
-          container,
-          textElement as ExcalidrawTextElementWithContainer,
-        ),
-      );
-    }
```

Comment:
one last thing: this seems a significant change?

Reply:
I don't think so. Before, the arrow label position was calculated during render, and now it is calculated right after the label change additionally. This is needed for the test env, because previously arrow labels didn't update their positions after resizing.

Reply:
> I don't think so. Before, the arrow label position was calculated during render, and now it is calculated right after the label change additionally.

We are actually trying to move all bound elements positioning to be done inside render. This will allow us to make render perf optimizations.

Reply:
in renderElement we do it in two places.

1. when we export:

https://github.com/excalidraw/excalidraw/blob/26ff3993bb0600cc3dbd77a8468820394a2204fc/src/renderer/renderElement.ts#L698-L761

3. and when we render on canvas during editing:

https://github.com/excalidraw/excalidraw/blob/26ff3993bb0600cc3dbd77a8468820394a2204fc/src/renderer/renderElement.ts#L436-L505

Reply:
It works fine. I just forgot what Aakansha mentioned once in Discord - you should use `LinearElementEditor.getBoundTextElementPosition` to get the updated position, the x,y in the element object are stale.